modules/data.remote/R/call_MODIS.R

```diff
@@ -14,78 +14,133 @@
+call_MODIS <- function(outfolder = "", start_date, end_date, lat, lon, size = 0, iter = i, product, band, band_qc = "", band_sd = "", siteID = "", package_method = "MODISTools", QC_filter = F, progress = T) {
   
   # makes the query search for 1 pixel and not for rasters for now. Will be changed when we provide raster output support.
   size <- 0
+  require(PEcAn.all)
   
-  # set start and end dates to correct format
-  if (package_method == "MODISTools"){
+  # reformat start and end date if they are in YYYY/MM/DD format instead of YYYYJJJ
+  if (grepl("/", start_date) == T)
+  {
+    start_date = as.Date(paste0(lubridate::year(start_date), spatial.tools::add_leading_zeroes(lubridate::yday(start_date), 3)), format = "%Y%j")
+  }
+  
+  if (grepl("/", end_date) == T)
+  {
+    end_date = as.Date(paste0(lubridate::year(end_date), spatial.tools::add_leading_zeroes(lubridate::yday(end_date), 3)), format = "%Y%j")
+  }
+  start_date = as.Date(start_date, format = "%Y%j")
+  end_date = as.Date(end_date, format = "%Y%j")
+  
+  #################### if package_method == MODISTools option ####################
+  if (package_method == "MODISTools")
+  {
     
+    #################### FUNCTION PARAMETER PRECHECKS #################### 
+    #1. check that modis product is available
     products = MODISTools::mt_products()
     if (!(product %in% products$product))
     {
       print(products)
       stop("Product not available for MODIS API. Please chose a product from the list above.")
-    } else {
-      print("Check #1: Product exists!")
-    }
-    
-
-    dates <- MODISTools::mt_dates(product = product, lat = lat, lon = lon)$modis_date
-    dates <- as.numeric(substr(dates, 2, nchar(dates)))
-    if (as.numeric(start_date) <= dates[1] | as.numeric(end_date) >= dates[length(dates)])
-    {
-      print(paste("Range of dates for product are ", dates[1], " - ", dates[length(dates)], sep = ""))
-      stop("Please choose dates between the date range listed above.")
-    } else {
-      print("Check #2: Dates are available!")
-    }
+    } 
     
+    #2. check that modis produdct band is available
     bands <- MODISTools::mt_bands(product = product)
     if (!(band %in% bands$band))
     {
       print(bands$band)
       stop("Band selected is not avialable. Please selected from the bands listed above that correspond with the data product.")
-    } else {
-      print("Check #3: Band Exists!")
+    } 
+    
+    #3. check that dates asked for in function parameters are fall within dates available for modis product/bands.
+    dates <- MODISTools::mt_dates(product = product, lat = lat, lon = lon)$modis_date
+    dates <- as.Date(as.character(substr(dates, 2, nchar(dates))), format = "%Y%j")
+    
+    ########## Date case 1:  user only wants one date ##########
+    if (start_date == end_date)
+    {
+      if (as.numeric(start_date) >= dates[1] & as.numeric(end_date) <= dates[length(dates)])
+      {
+        print("Extracting data")
```

Comment:
In general, we should be using `PEcAn.logger` functions for messages.

FWIW, in standalone code, stuff like this is better passed through `message` rather than `print` because the former is easier to suppress in higher-level code.

Reply:
```suggestion
        PEcAn.logger::logger.info("Extracting data")
```